src/edu/washington/escience/myria/operator/SymmetricHashJoin.java

```diff
@@ -331,7 +331,13 @@ public SymmetricHashJoin(final List<String> outputColumns, final Operator left,
   @Override
   protected Schema generateSchema() {
     final Schema leftSchema = getLeft().getSchema();
+    if (leftSchema == null) {
```

Comment:
OK, did you only modify SymmetricHashJoin because that's the only streaming join operator we have? Could other operators have children with dynamic schema in the future?